gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/authz/AbstractAuthorizer.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.server.authz;
+
+import org.apache.tinkerpop.gremlin.driver.Tokens;
+import org.apache.tinkerpop.gremlin.driver.message.RequestMessage;
+import org.apache.tinkerpop.gremlin.process.computer.traversal.strategy.decoration.VertexProgramStrategy;
+import org.apache.tinkerpop.gremlin.process.traversal.Bytecode;
+import org.apache.tinkerpop.gremlin.process.traversal.step.LambdaHolder;
+import org.apache.tinkerpop.gremlin.process.traversal.util.BytecodeUtil;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticatedUser;
+import org.apache.tinkerpop.gremlin.structure.io.graphson.GraphSONMapper;
+import org.apache.tinkerpop.gremlin.structure.io.graphson.GraphSONVersion;
+import org.apache.tinkerpop.shaded.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+
+
+/**
+ * Provides utilities for implementing the {@link Authorizer} interface.
+ *
+ * @author Marc de Lignie
+ */
+public abstract class AbstractAuthorizer implements Authorizer{
+    private static final ObjectMapper mapper = GraphSONMapper.build().version(GraphSONVersion.V2_0).create().createMapper();
+
+    /**
+     * Authorizes a user for a request.
+     *
+     * @param user {@link AuthenticatedUser} to be used for the authorization
+     * @param msg RequestMessage to authorize the user for
+     */
+    public void authorize(final AuthenticatedUser user, final RequestMessage msg) throws AuthorizationException {
+        switch (msg.getOp()) {
+            case Tokens.OPS_EVAL:
+                final String script = (String) msg.getArgs().get(Tokens.ARGS_GREMLIN);
+                authorizeString(user, script, msg);
+                break;
+            case Tokens.OPS_BYTECODE:
+                final Bytecode bytecode;
+                try {
+                    final Object bytecodeObj = msg.getArgs().get(Tokens.ARGS_GREMLIN);
+                    bytecode = bytecodeObj instanceof Bytecode ? (Bytecode) bytecodeObj :
+                            mapper.readValue(bytecodeObj.toString(), Bytecode.class);
```

Comment:
Under what circumstance does it happen that we get a `OPS_BYTECODE` but its value remains a GraphSON 2.0 string that needs to be deserialized? if so, i would think we would want to figure out how to not let that happen and keep deserialization details out of the `Authorizer`?

Reply:
I will look into this, probably an oversight or some old test configs.